### ISSUE TITLE
Add aerosol support

### ISSUE THREAD

**Issue by EdGrrr** [EdGrrr/pyLRT#1]
It should be pretty simple to add support for varying aerosols, using the same design as the cloud file.

**Comment by EdGrrr** [comment]
For anyone who has arrived here thinking 'where is the aerosol support?' this is the place for you!

I do not currently use aerosols with libradtran, so I am not clear what aerosol support should look like. So this is where I need your help. What should the aerosol support look like? How would you interact with it? What features are necessary?

I can't promise I will add them all immediately, but if you ave any suggestions, please let me know.

**Comment by deepthi-narasimhan** [comment]
Hello @EdGrrr , 
Firstly, thank you for the succinct and well done open source code. I'm currently working with aerosols and am intending to use your code for that. 
A function that might help immensely with aerosol support is the ability to write aerosol files (example: aerosol expilict optical files) and weight them together by relative abundances using python. With libradtran, any subsequently defined aerosol file set overrides the previous profile. The only way to mix them is by defining them as aerosol species files (like within the OPAC subdirectory in libradtran source installations) and then weight them together by mass concentration abundances at vertical profile levels. Writing new files into these files would be a huge advantage especially when working with these files recursively. In my work, for example, my aerosol is not defined in the HITRAN database, so externally computed spectrally dependent extinction coeff, ssa, and g are all I have. So far, the only way to work with this that I can think of is using aerosol file explicit - as it preserves spectral dependence. The issue with it is if I define this, all other aerosols are obliterated (ref. overwriting of prev. aerosol files). I would love to be able to write aerosol species files at each grid point preserving spectral dependence, and if pylrt could do that, I'm sure its application would be wonderful.

**Comment by EdGrrr** [comment]
That @deepthi-narasimhan - that is really helpful. Just so I have this clear, you would like

- The ability to write 'aerosol files', as defined in the '_aerosol_file_' input parameter (based on profiles of gg, ssa etc.). Ideally this would also include the '_explicit_' option, which is a set of two different files (one overall 'profile' and a set of files that define the spectral properties at each level).

That should be possible to add in, although it might take a little while to find some time for it. I am happy to look at a pull request if you want.

There is some framework do do this already in pyLRT that you could use (there is some code to write the cloud files which I think could be re-purposed for aerosol files). For the 'explicit' seutp, it would then also be necessary to keep track of the names of all the relevant temporary files.

Similar code could also be used to support the '_aerosol_species_file_' parameter too

**Comment by deepthi-narasimhan** [comment]
@EdGrrr yes, however what I was thinking more about is being able to mix aerosols with the aerosol explicit through the wrapping code too. The aerosol file explicit thing would do it for one aerosol that we define. In lrt, from what I've understood of the documentation, defining a different kind of aerosol file type input overrides the previous ones, except for within the option related to aerosol species file wjere we can declare the mixing explicitly. The issue though is that declaring aerosol file explicit would , from what I've understood according to lrt documentation, override the aerosol species inputs. Effectively, my aerosol file explicit ghg ssa and tau has to be premixed for the whole set of aerosols in the atmosphere. 
Two ways we could maybe look at to go about this are pylrt codes that either can put in an aerosol file explicit as input in the opac subdir as another new aerosol species, and then calling uvspec with the aerosol species option with us defining relative abundances, or one that does a mixing of existing aerosols and our predefined ones to make aer explicit file ghg ssa and tau files for the whole atmosphere and passing them as inputs. I'm not sure if this is clearer, please let me know if you'd like me to rephrase.
I did look at repurposing the cloud file for this, but I'm still quite unsure how to incorporate this into that framework. 
Yes please, do you think a pull request could be done?
